neurokit2/ecg/ecg_findpeaks.py

```diff
@@ -1051,6 +1053,80 @@ def _ecg_findpeaks_rodrigues(signal, sampling_rate=1000, **kwargs):
     return rpeaks
 
 
+# =============================================================================
+# Visibility graph transformation - by Koka and Muma (2022)
+# =============================================================================
+def _ecg_findpeaks_vgraph(signal, sampling_rate=1000, lowcut=3, order=2, **kwargs):
+    """R-Peak Detector Using Visibility Graphs by Taulant Koka and Michael Muma (2022).
+
+    References
+    ----------
+    - T. Koka and M. Muma (2022), Fast and Sample Accurate R-Peak Detection for Noisy ECG Using
+      Visibility Graphs. In: 2022 44th Annual International Conference of the IEEE Engineering
+      in Medicine & Biology Society (EMBC). Uses the Pan and Tompkins thresholding.
+
+    """
+    # Try loading ts2vg
+    try:
+        from ts2vg import NaturalVG
+    except ImportError as import_error:
+        raise ImportError(
+            "NeuroKit error: ecg_findpeaks(): the 'ts2vg' module is required for"
+            " this method to run. Please install it first (`pip install ts2vg`)."
+        ) from import_error
+
+    # Filter signal using highpass Butterworth filter
+    nyq = 0.5 * sampling_rate
+    high = lowcut / nyq
+    b, a = scipy.signal.butter(order, high, btype="highpass")
+    filtered = scipy.signal.filtfilt(b, a, signal)
```

Comment:
Should we filter the signal again or do we assume that this cleaning should be done beforehand (using `ecg_clean()`) and the signal provided should be clean?

Reply:
Well, I did it like for example with the Kalidas, Christov, Zong, etc. detectors where nothing is filtered in ecg_clean() and the raw signal is passed on. This means that there is no re-filtering and filtering is only part of the detector.

Reply:
@sokolmarek Unless this is based on an existing implementation (I saw the paper mentioned that the Python code would be made publicly available on GitHub but I couldn't find it), I think it would be better to add a new method to `ecg_clean()` using the `"butterworth"` method of `signal_filter()`

Reply:
And @DominiqueMakowski I also think we should consider doing the same for some of the other detectors, at least Zong where it is also a butterworth filter: https://github.com/neuropsychology/NeuroKit/blob/5c749d514ad9c9d5fd6d65c39c9a101e1eb3e4cf/neurokit2/ecg/ecg_findpeaks.py#L476-L482

However, the `"butterworth"` method of `signal_filter()` uses zero-phase filtering with [`scipy.signal.sosfiltfilt()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.sosfiltfilt.html) whereas the current Zong implementation uses [`scipy.signal.lfilter()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.lfilter.html), not sure if another method should be added to `signal_filter()` so that no functional changes are made?

Reply:
> we should consider doing the same for some of the other detectors, at least Zong where the cleaning is also just a butterworth filter

Yes that makes sense, as otherwise we risk doing double cleaning and cannot dissociate the effect of cleaning method vs peak detection per se

Reply:
@danibene So I added a new `ecg_clean()` method using `signal_filter()` and removed the original filtering from the detector. At the same time, I think it's a good idea to remove the other filtering in other detectors in the future. Signal preprocessing is often individual and one has to play with it sometimes to design the right filter. There is no gold standard. The fact that such filtering worked for their ECG recordings within the Zong detector does not mean that it will be as effective elsewhere.

Reply:
> So I added a new ecg_clean() method using signal_filter() and removed the original filtering from the detector.

@sokolmarek Thank you! `signal_filter()` takes `lowcut` in Hz so you can remove the normalization step and directly use `lowcut=4`. Also, is there a reason you want to use `"butterworth_ba"` instead of `"butterworth"` (which uses [`scipy.signal.sosfiltfilt()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.sosfiltfilt.html))? I think we have generally had better results with the latter

Reply:
> > So I added a new ecg_clean() method using signal_filter() and removed the original filtering from the detector.
> 
> @sokolmarek Thank you! `signal_filter()` takes `lowcut` in Hz so you can remove the normalization step and directly use `lowcut=4`. Also, is there a reason you want to use `"butterworth_ba"` instead of `"butterworth"` (which uses [`scipy.signal.sosfiltfilt()`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.signal.sosfiltfilt.html))? I think we have generally had better results with the latter

@danibene All right, I'll edit it:) Since there was no mention of using the second-order section in the article as part of the filtering, I didn't use it here either. I'll rewrite it to sos, but I think in this low order filter it won't have that much effect. At least it minimizes the ringing effect on the edges I guess...

Reply:
@sokolmarek thank you :) I'll go ahead and merge this now, I think any changes to the other peak detection functions e.g., adding a new `ecg_clean()` method for the Zong detector, should be done in a separate PR